flow/util/genReport.py

```diff
@@ -0,0 +1,287 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+
+# make sure the working dir is flow/
+os.chdir(os.path.join(os.path.dirname(os.path.abspath(__file__)) , '..'))
+
+reportFilename = 'report.log'
+singleReportFilename = 'reports/' + reportFilename
+drcFilename = '5_route_drc.rpt'
+lastExpectedLog = '6_report.log'
+metricsLogFmt = 'gen-metrics-{}-check.log'
+metricsCheckFmt = 'metadata-{}-check.log'
+regexError = re.compile(r"^\[?err", re.VERBOSE | re.IGNORECASE)
+regexWarning = re.compile(r"^\[?warn", re.VERBOSE | re.IGNORECASE)
+
+helpText = '''
+Scans "./logs" and "./reports" folders for errors and warnings.
+'''
+parser = argparse.ArgumentParser(description=helpText)
+
+helpText = '''
+Level 0: only report fail/pass (default).
+Level 1: report count per file.
+Level 2: report full messages.
+'''
+parser.add_argument('--verbose', '-v',
+                    required=False, default=0, action='count',
+                    help=helpText)
+
+parser.add_argument('--file', '-f',
+                    required=False, default=False, action='store_true',
+                    help='Generate final report file.')
+
+parser.add_argument('--no-print', '-q',
+                    required=False, default=False, action='store_true',
+                    help='Do not print report to stdout.')
+
+parser.add_argument('--single', '-s',
+                    required=False, default=False, action='store_true',
+                    help='Generate aggregated file with all reports.')
+
+args = parser.parse_args()
+verbose = args.verbose
+generateSingleFile = args.single
+generateFile = args.file
+printOutput = not args.no_print
+
+
+def parseMessages(filename):
+    errors = list()
+    warnings = list()
+    try:
+        with open(filename, 'r') as f:
+            lines = f.readlines()
+    except Exception as e:
+        msg = 'Failed to open ' + filename + '.'
+        msg += ' Check to see if design finished.'
+        print(msg)
+        return [msg], [msg]
+    for line in lines:
+        if re.search(regexError, line):
+            errors.append(line.strip())
+        if re.search(regexWarning, line):
+            warnings.append(line.strip())
+    return errors, warnings
+
+
+def finish(output, outputFile='', summary=False):
+    global generateSingleFile
+    global generateFile
+    global printOutput
+    global reportFilename
```

Comment:
I don't see any writes to these variables so I don't think these are needed.

Reply:
Fixed on 9947443.